docs: add expose ports section to Jobs configuration guide (#2551)
huggingface/hub-docs: docs/hub/jobs-configuration.md

l40sx8           8x Nvidia L40S          192 vCPU  1534 GB  6500 GB   8x L40S (384 GB)          $0.3917   $23.50
```

## Expose Ports

> Jobs can expose container ports through the public jobs proxy using `--expose <port>` (CLI) or `expose=[<port>]` (Python API). Each exposed port is reachable at `https://<job_id>--<port>.hf.jobs` and requires an HF token with `read` access to the job's namespace.

Reviewer: Tested against a live job: the token must go in the `Authorization` header — query params and cookies don't work, and browser-style requests get a bare JSON 401 (no login redirect). Since pasting the URL into a browser is the first thing most people will try, worth showing the auth concretely:

````suggestion
Jobs can expose container ports through the public jobs proxy using `--expose <port>` (CLI) or `expose=[<port>]` (Python API). Each exposed port is reachable at `https://<job_id>--<port>.hf.jobs` and requires an HF token with `read` access to the job's namespace, passed in the `Authorization` header:

```bash
curl -H "Authorization: Bearer $HF_TOKEN" https://<job_id>--<port>.hf.jobs/
```

This is the only supported authentication method — exposed URLs can't be opened directly in a browser.
````

Reviewer: > This is the only supported authentication method — exposed URLs can't be opened directly in a browser.

I didn't try and get this working super hard but Fable failed to do this with a few attemtps

Reviewer 2: +1, though it could be cool in the future to support cookies, e.g. for user-defined dashboards monitoring the job or things like that

Author: ````suggestion
Jobs can expose container ports through the public jobs proxy using `--expose <port>` (CLI) or `expose=[<port>]` (Python API). Each exposed port is reachable at `https://<job_id>--<port>.hf.jobs` and requires an HF token with `read` access to the job's namespace:

```bash
curl -H "Authorization: Bearer $HF_TOKEN" https://<job_id>--<port>.hf.jobs/
```
````

let's put a curl example yes. The cookie auth shouldn't be too difficult to add

Reviewer 3: Basic-Auth probably more than Cookie

> This works on `hf jobs run`, `hf jobs uv run`, and their scheduled variants.

Reviewer: The flag is repeatable, but that's easy to miss:

```suggestion
This works on `hf jobs run`, `hf jobs uv run`, and their scheduled variants. Repeat the flag to expose multiple ports (`--expose 8000 --expose 8001`), or pass several ports in the list (`expose=[8000, 8001]`).
```

> [!NOTE]

> > Exposing ports is a paid service. See the [pricing page](./jobs-pricing) for details on exposed port billing.

Reviewer: "A paid service" reads a bit like a plan gate. My first question was whether this needs a paid plan, or is a surcharge on top of the job. Suggest stating it's additive to the hardware price:

```suggestion
> Exposed ports are billed at a small flat hourly rate on top of the job's hardware price, only while the job is running. See the [pricing page](./jobs-pricing) for details.
```

Reviewer: Maybe a bit clearer?
